lua/telescope/builtin/init.lua

```diff
@@ -21,58 +30,248 @@ end
 builtin.filetypes = require('telescope.builtin.internal').filetypes
+
+--- Lists all available highlights
+---@param opts table: options to pass to the picker
 builtin.highlights = require('telescope.builtin.internal').highlights
+
+--- Lists vim autocommands and goes to their declaration on <cr>
+---@param opts table: options to pass to the picker
 builtin.autocommands = require('telescope.builtin.internal').autocommands
+
+--- Lists spelling suggestions for the current word under the cursor, replaces word with selected suggestion on <cr>
+---@param opts table: options to pass to the picker
 builtin.spell_suggest = require('telescope.builtin.internal').spell_suggest
+
+--- Lists the tag stack for the current window
+---@param opts table: options to pass to the picker
 builtin.tagstack = require('telescope.builtin.internal').tagstack
 
+
+--------------------------------------------------
+--
+-- LSP-related Pickers
+--
+--------------------------------------------------
+--- Lists LSP references for word under the cursor
+---@param opts table: options to pass to the picker
+---@field shorten_path boolean: if true, will shorten path shown
 builtin.lsp_references = require('telescope.builtin.lsp').references
+
+--- Goto the definition of the word under the cursor, if there's only one, otherwise show all options in Telescope
+---@param opts table: options to pass to the picker
 builtin.lsp_definitions = require('telescope.builtin.lsp').definitions
+
+--- Goto the implementation of the word under the cursor if there's only one, otherwise show all options in Telescope
+---@param opts table: options to pass to the picker
 builtin.lsp_implementations = require('telescope.builtin.lsp').implementations
+
+--- Lists LSP document symbols in the current buffer
+---@param opts table: options to pass to the picker
+---@field ignore_filename type: string with file to ignore
 builtin.lsp_document_symbols = require('telescope.builtin.lsp').document_symbols
+
+--- Lists any LSP actions for the word under the cursor, that can be triggered with <cr>
+---@param opts table: options to pass to the picker
 builtin.lsp_code_actions = require('telescope.builtin.lsp').code_actions
-builtin.lsp_document_diagnostics = require('telescope.builtin.lsp').diagnostics
-builtin.lsp_workspace_diagnostics = require('telescope.builtin.lsp').workspace_diagnostics
+
+--- Lists any LSP actions for a given range, that can be triggered with <cr>
+---@param opts table: options to pass to the picker
 builtin.lsp_range_code_actions = require('telescope.builtin.lsp').range_code_actions
+
+--- Lists LSP document symbols in the current workspace
+---@param opts table: options to pass to the picker
+---@field shorten_path boolean: if true, will shorten path shown
+---@field ignore_filename string: file(s) to ignore
 builtin.lsp_workspace_symbols = require('telescope.builtin.lsp').workspace_symbols
+
+--- Lists LSP for all workspace symbols asynchronously
```

Comment:
I think `asynchronously` sounds a bit ambiguous here. It *is* async but it doesn't tell you if you can 'refresh' the result on-the-fly. `dynamically` or `live` is a better word IMO because it's similar to `live_grep`.